src/chat.rs

```diff
@@ -4899,6 +4993,30 @@ impl Context {
                     .await?
                     .id
             }
+            SyncId::ContactFingerprint(fingerprint) => {
+                let name = "";
+                let addr = "";
+                let (contact_id, _) =
+                    Contact::add_or_lookup_ex(self, name, addr, fingerprint, Origin::Hidden)
+                        .await?;
+                match action {
+                    SyncAction::Rename(to) => {
+                        contact_id.set_name_ex(self, Nosync, to).await?;
+                        self.emit_event(EventType::ContactsChanged(Some(contact_id)));
+                        return Ok(());
+                    }
+                    SyncAction::Block => {
+                        return contact::set_blocked(self, Nosync, contact_id, true).await
+                    }
```

Comment:
Just a non-actionable remark: It's a bit weird that this calls `contact::set_blocked(.., true)`, and then a few lines below, `chat_id.block_ex()` is called, which again calls `contact::set_blocked(.., true)`. But how (un)blocking a chat and (un)blocking a contact interacts is just super weird anyways, maybe we can clean that up sometime. Altough, not sure if anyone really uses the "block" feature, a spammer can just create a new account anyways, and if you want a chat to be out of your view you can archive&mute it.

Doing nothing here, and relying on `chat_id.block.ex()` to also block the contact, would be even weirder, so, I'm not proposing to change anything here.